### PR TITLE
feat: add quality tests

### PR DESCRIPTION
**Pull request by JuanDavidBuitrago**
Description:
This PR adds:
- QA to maintain the repo.
- Routes folder with each endpoint configuration.

The routes are configured with blueprint of Flask. The previous routes were moved to and config to the routes directory

### CODE REVIEW

**Comment by mariajgrimaldi** [comment]
There are far more changes than you're describing. Can you add them to the description?

**Comment by mariajgrimaldi** [comment]
There are far more changes than you're describing. Can you add them to the description?

**Comment by JuanDavidBuitrago** [comment]
> There are far more changes than you're describing. Can you add them to the description?

@mariajgrimaldi I add the changes in description, maybe you consider it should be in other PR?

The additional changes are in routes directory, it contains endpoints configurations

**Comment by mariajgrimaldi** [comment]
Yes, I would like to see those changes in another PR with tests included so we know 100% are working.

File: codejailservice/tests/test_app.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+APP_ROUTE = "http://localhost:8550"
+
+
+class CodejailserviceTest(unittest.TestCase):
+    """
+    Test case for Codejail Service
+    """
+
+    def test_works(self):
+        assert True
```
Comment (mariajgrimaldi) [inline codejailservice/tests/test_app.py:15]: please remove this test

File: codejailservice/tests/test_app.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    def test_main_route(self):
+        """This run that the main route app is working and sending a text."""
+        response = requests.get(APP_ROUTE)
+
+        assert response.status_code == 200
+        assert isinstance(response.text, str)
```
Comment (mariajgrimaldi) [inline codejailservice/tests/test_app.py:22]: what does it mean to test the main route?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    - name: Set up Python 3.8
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+    - name: Install dependencies
+      run: |
+        make requirements
+    - name: Run the QA suite
+      run: |
+        make quality
```
Comment (mariajgrimaldi) [inline .github/workflows/ci.yml:29]: add `make run && make test`

File: codejailservice/tests/test_app.py (hunk 2 above)
Comment (mariajgrimaldi) [reply codejailservice/tests/test_app.py:22]: check what the text contains

File: codejailservice/tests/test_app.py [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+    def test_health_route(self):
+        """This run that the health route  app is working and sending a text."""
+        response = requests.get(f"{APP_ROUTE}/health")
+
+        assert response.status_code == 200
+        assert isinstance(response.text, str)
```
Comment (mariajgrimaldi) [inline codejailservice/tests/test_app.py:29]: check what it contains

File: .github/workflows/ci.yml (hunk 3 above)
Comment (mariajgrimaldi) [reply .github/workflows/ci.yml:29]: Correction: I'm doing this in PR #11

File: codejailservice/tests/test_app.py (hunk 1 above)
Comment (mariajgrimaldi) [reply codejailservice/tests/test_app.py:15]: Correction: I'm doing this in PR #11

File: codejailservice/tests/test_app.py (hunk 2 above)
Comment (mariajgrimaldi) [reply codejailservice/tests/test_app.py:22]: Correction: I'm doing this in PR #11

File: codejailservice/tests/test_app.py (hunk 4 above)
Comment (mariajgrimaldi) [reply codejailservice/tests/test_app.py:29]: Correction: I'm doing this in PR #11

**Review by mariajgrimaldi — APPROVED** [review]

**Pull request merged**